### ISSUE TITLE
Allow support for replacing complete PubSub list of NULLs with a single NULL (3.5.6.8m MAL BB)

### ISSUE THREAD

**Issue by CesarCoelho** [SamCooper/JAVA_SPEC_RIDS#23]
Currently, it is possible to send a null on the update value list when a message is published.
MAL Java API should throw an exception when this happens. Please fix the implementation.

**Comment by SamCooper** [comment]
This should be raised on the relevant implementation if it is a bug on an implementation, this issue list for issues with the Java API specification.

However, I actually think that this IS an issue for the MAL (and therefore Java API) specification in so much that it should allow you to replace a complete list with NULL if and only if you are wanting to send a list of NULLs.

Therefore I have renamed this issue appropriately and raised the correct issue on the MAL specification:

https://github.com/SamCooper/MAL_SPEC_RIDS/issues/7

**Comment by SamCooper** [comment]
If this is "fixed" in the MAL Blue book it should be reflected here, otherwise the MAL Java API should throw an exception if passed a NULL here.

**Comment by CesarCoelho** [comment]
Hi Sam, what I understand from reading both issues, is that you closed the previous one because it was repeated. (Sam: "This is actually an issue with the MAL and Java API")
It doesn't seem to me that it was "fixed".

I don't remember the exact details but I think that I found this problem when I started using DLR's implementation of the SPP enconding and we concluded that DLR's behaviour was the correct one but in fact, the error should be thrown on the layer above, the MAL.

By the way, can you start linking the fixed issue to the corresponding git commit? Everything will become neater that way

**Comment by SamCooper** [comment]
No, I meant that if it is fixed in the MAL Blue book then the Java API needs to be updated to reflect that new way of working, if it is not fixed in the blue book, i.e. the blue book is unchanged, then we should change the Java API to throw an exception.

These issues BTW are on the specifications not the implementations, although I can probably link to a git commit for the ESA implementation if there is one.
